httpx/_urls.py

```diff
@@ -74,7 +74,7 @@ class URL:
       themselves.
     """
 
-    def __init__(self, url: URL | str = "", **kwargs: typing.Any) -> None:
+    def __init__(self, url: URL | str | typing.Any = "", **kwargs: typing.Any) -> None:
```

Comment:
This will allow any type, which makes the other 2 types in this union irrelevant.

```suggestion
    def __init__(self, url: URL | str = "", **kwargs: typing.Any) -> None:
```

Maybe what you want is something like this?
```py
class HasDunderStr(Protocol):
    def __str__(self): ...
```

But... Not sure this is useful.

Reply:
I quite like keeping the first two types. It makes it explicit that there is a great compatibility for them.

I like your idea to use protocol. I'm gonna implement it right away.

Reply:
Note the "But... Not sure this is useful." 👀 👍